src/main/java/com/epam/rd/autotasks/springemployeecatalog/rowpappers/ManagerResultSetExtractor.java

```diff
@@ -0,0 +1,49 @@
+package com.epam.rd.autotasks.springemployeecatalog.rowpappers;
+
+import com.epam.rd.autotasks.springemployeecatalog.domain.Department;
+import com.epam.rd.autotasks.springemployeecatalog.domain.Employee;
+import com.epam.rd.autotasks.springemployeecatalog.services.ExtractorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.stereotype.Component;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.epam.rd.autotasks.springemployeecatalog.AppConstants.ALL_FROM_DEPARTMENT;
+import static com.epam.rd.autotasks.springemployeecatalog.AppConstants.DEPARTMENT;
+import static com.epam.rd.autotasks.springemployeecatalog.AppConstants.ID;
+
+@Component
+public class ManagerResultSetExtractor implements ResultSetExtractor<Map<Long, Employee>> {
+    private final ExtractorService extractorService;
+    private final JdbcTemplate jdbcTemplate;
+    private final DepartmentResultSetExtractor departmentResultSetExtractor;
+
+    @Autowired
+    public ManagerResultSetExtractor(ExtractorService extractorService,
+                                     JdbcTemplate jdbcTemplate,
+                                     DepartmentResultSetExtractor departmentResultSetExtractor) {
+        this.extractorService = extractorService;
+        this.jdbcTemplate = jdbcTemplate;
```

Comment:
Extractor should not aware of jdbcTemplate. This violates the Single Responsibility Principle.
Extractor should just extract Entity from Result set